src/form/bf_mx.jl

```diff
@@ -388,3 +193,528 @@ function constraint_tp_model_voltage_magnitude_difference(pm::_PMs.GenericPowerM
+    end
+    # create vectors of the diagonal elements
+    ncnds = length(_PMs.conductor_ids(pm, nw))
+    diag_re = Dict([(id, [_PMs.var(pm, nw, c, :pg, id) for c in 1:ncnds]) for id in gen_ids])
+    diag_im = Dict([(id, [_PMs.var(pm, nw, c, :qg, id) for c in 1:ncnds]) for id in gen_ids])
+    # calculate bounds for matrix variable
+    bound = Dict{eltype(gen_ids), Array{Real,2}}()
+    for (id, gen) in _PMs.ref(pm, nw, :gen)
+        bus_id = gen["gen_bus"]
+        vmax = _PMs.ref(pm, nw, :bus, bus_id)["vmax"].values
+        vmin = _PMs.ref(pm, nw, :bus, bus_id)["vmin"].values
+        pmax = max(abs.(gen["pmax"].values), abs.(gen["pmin"].values))
+        qmax = max(abs.(gen["qmax"].values), abs.(gen["qmin"].values))
+        smax = sqrt.(pmax.^2 + qmax.^2)
+        cmax = smax./vmin
+        bound[id] = vmax*cmax'
+    end
+    # create matrix variables, whilst injecting diagonals
+    (Pg,Qg) = variable_mx_complex_with_diag(pm.model, gen_ids, ncnds, bound;
+        diag_re=diag_re, diag_im=diag_im, name=("Pg", "Qg"), prefix="$nw")
+    # save references
+    _PMs.var(pm, nw)[:Pg] = Pg
+    _PMs.var(pm, nw)[:Qg] = Qg
+end
+
+
+function variable_tp_generation_current_mx(pm::_PMs.GenericPowerModel; nw=pm.cnw)
+    gen_ids = collect(_PMs.ids(pm, nw, :gen))
+    ncnds = length(_PMs.conductor_ids(pm, nw))
+    # calculate bounds
+    bound = Dict{eltype(gen_ids), Array{Real,2}}()
+    for (id, gen) in _PMs.ref(pm, nw, :gen)
+        bus = _PMs.ref(pm, nw, :bus, gen["gen_bus"])
+        cmax = _gen_curr_max(gen, bus)
+        bound[id] = cmax*cmax'
+    end
+    # create matrix variables
+    (CCgr,CCgi) = variable_mx_hermitian(pm.model, gen_ids, ncnds, bound; name="CCg", prefix="$nw")
+    # save references
+    _PMs.var(pm, nw)[:CCgr] = CCgr
+    _PMs.var(pm, nw)[:CCgi] = CCgi
+end
+
+
+function variable_tp_load_mx(pm::_PMs.GenericPowerModel; nw=pm.cnw)
+    load_wye_ids = [id for (id, load) in _PMs.ref(pm, nw, :load) if load["conn"]=="wye"]
+    load_del_ids = [id for (id, load) in _PMs.ref(pm, nw, :load) if load["conn"]=="delta"]
+    load_cone_ids = [id for (id, load) in _PMs.ref(pm, nw, :load) if _load_needs_cone(load)]
+    # first create the wye loads; will create keys :Pd, :Qd
+    variable_tp_load_power_wye_mx(pm, load_wye_ids)
+    # now, create delta loads; will create :Xdr, :Xdi
+    variable_tp_load_power_delta(pm, load_del_ids)
+    # define :Pd, :Qd for delta loads as lin. transformation of :Xdr and :Wdim
+    Td = [1 -1 0; 0 1 -1; -1 0 1]
+    for id in load_del_ids
+        _PMs.var(pm, nw, :Pd)[id] = _PMs.var(pm, nw, :Xdr, id)*Td
+        _PMs.var(pm, nw, :Qd)[id] = _PMs.var(pm, nw, :Xdi, id)*Td
+    end
+    # both loads need a current variable; bounds adjusted for connection type
+    variable_tp_load_current_mx(pm, [load_wye_ids..., load_del_ids...])
+    variable_tp_load_power_vector(pm, load_cone_ids)
+end
+
+
+function variable_tp_load(pm::_PMs.GenericPowerModel{T}; nw=pm.cnw) where T <: AbstractUBFForm
```

Comment:
Is this function matrix specific? should it have an additional `mx` suffix to differentiate it? check other functions without `mx` suffix as well

Reply:
Functions with the prefix `mx` are functions related to the matrix KCL formulation; this function can be used for both the matrix KCL and the ordinary KCL we used so far, so in the current use of `mx`, we do not need it as a prefix here.